GTMAppAuthSwift/Sources/KeychainStore.swift

```diff
@@ -79,6 +79,8 @@ public final class KeychainStore: NSObject {
     self.itemName = itemName
     self.keychainAttributes = keychainAttributes
     self.keychainHelper = keychainHelper
+
+    NSKeyedUnarchiver.setClass(AuthState.self, forClassName: "GTMAppAuthFetcherAuthorization")
```

Comment:
An alternate approach here would be to do the reverse and use setName:forClass: on `NSKeyedArchiver` to maintain the old class name `GTMAppAuthFetcherAuthorization` for our keychain storage.  One benefit being that we would preserve backward compatibility.

Reply:
I think either way we'd need to do the name migration on unarchive to handle cases where older versions of the library archived auth state. I do like the idea of setting the migration for archiving so that older versions of the library can work with the new type.